packages/python/plotly/plotly/express/_core.py

```diff
@@ -754,6 +754,202 @@ def apply_default_cascade(args):
+        reserved_names = _initialize_argument_col_names(
+            args, attrables, array_attrables
+        )
+    else:
+        reserved_names = set()
+
+    # Case of functions with a "dimensions" kw: scatter_matrix, parcats, parcoords
+    if "dimensions" in args and args["dimensions"] is None:
+        if not df_provided:
+            raise ValueError(
+                "No data were provided. Please provide data either with the `data_frame` or with the `dimensions` argument."
+            )
+        else:
+            df_args = args["data_frame"]
+            df[df_args.columns] = df_args[df_args.columns]
+
+    # Loop over possible arguments
+    for field_name in attrables:
+        # Massaging variables
+        argument_list = (
+            [args.get(field_name)]
+            if field_name not in array_attrables
+            else args.get(field_name)
+        )
+        # argument not specified, continue
+        if argument_list is None or argument_list is [None]:
+            continue
+        # Argument name: field_name if the argument is not a list
+        # Else we give names like ["hover_data_0, hover_data_1"] etc.
+        field_list = (
+            [field_name]
+            if field_name not in array_attrables
+            else [field_name + "_" + str(i) for i in range(len(argument_list))]
+        )
+        # argument_list and field_list ready, iterate over them
+        # Core of the loop starts here
+        for i, (argument, field) in enumerate(zip(argument_list, field_list)):
+            length = len(df)
+            if argument is None:
+                continue
+            ## ----------------- argument is a col name ----------------------
+            elif isinstance(argument, str) or isinstance(
+                argument, int
+            ):  # just a column name given as str or int
+                if not df_provided:
+                    raise ValueError(
+                        "String or int arguments are only possible when a"
+                        "DataFrame or an array is provided in the `data_frame`"
+                        "argument. No DataFrame was provided, but argument '%s'"
+                        "is of type str or int." % field
+                    )
+                # Check validity of column name
+                if argument not in args["data_frame"].columns:
+                    raise ValueError(
+                        "Value of '%s' is not the name of a column in 'data_frame'. "
+                        "Expected one of %s but received: %s"
+                        % (field, str(list(args["data_frame"].columns)), argument)
+                    )
+                if length and len(args["data_frame"][argument]) != length:
+                    raise ValueError(
+                        "All arguments should have the same length."
+                        "The length of column argument `df[%s]` is %d, whereas the"
+                        "length of previous arguments is %d"
```

Comment:
is it possible to name the previous argument here? "previous" won't mean anything to the user as they don't know the iteration order